lib/rubocop_director/runner.rb

```diff
@@ -1,33 +1,63 @@
 require "optparse"
+require "optparse/date"
 
 require_relative "commands/generate_config"
 require_relative "commands/plan"
 
 module RubocopDirector
   class Runner
     def initialize(args)
-      arg_parser.parse(args)
-      @command ||= Commands::Plan.new(@since)
+      @options = {}
+      arg_parser.parse(args, into: @options)
+      verify_options
     end
 
     def perform
-      @command.run.either(
+      command.run.either(
         ->(success_message) { puts success_message },
         ->(failure_message) { puts "\nFailure: #{failure_message}" }
       )
     end
 
     private
 
+    def verify_options
+      @options[:rubocop_config] = verified_path(config_name: RUBOCOP_CONFIG_NAME, path: @options[:rubocop_config])
+      @options[:director_config] = verified_path(config_name: CONFIG_NAME, path: @options[:director_config])
+    end
+
+    def verified_path(config_name:, path:)
+      path = project_root if path.nil?
+      path.directory? ? path + config_name : path
+    end
+
+    def project_root
+      return Rails.root if defined?(Rails)
+      return Bundler.root if defined?(Bundler)
+
+      Pathname.new(Dir.pwd)
+    end
+
+    def command
+      @command ||= if @options[:generate_config]
+        Commands::GenerateConfig.new(**@options.slice(:director_config))
+      else
+        Commands::Plan.new(**@options.slice(:since, :director_config, :rubocop_config))
+      end
+    end
+
     def arg_parser
       OptionParser.new do |p|
-        p.on("--generate-config", "Generate default config based on .rubocop_todo.yml") do |since|
-          @command = Commands::GenerateConfig.new
+        p.accept(Pathname) do |s|
+          Pathname.new(s)
+        rescue ArgumentError, TypeError
+          raise OptionParser::InvalidArgument, s
         end
 
-        p.on("--since=SINCE", "Specify date to start checking git history") do |since|
-          @since = since
-        end
+        p.on("--generate_config", "Generate default config based on .rubocop_todo.yml")
```

Comment:
I'd rather keep `-` version everywhere (`--generate-config` instead of `--generate_config`)

Reply:
Hey, I did some testing when I decided to go with `-` instead which I forgot to mention.
`hyphen` (-) and `underscore` (_) can be used interchangeably when using them with `OptionParser`.

ie, `bundle exec rubocop-director --generate-config` == `bundle exec rubocop-director --generate-config`

The reason why I went ahead with an underscore is that args containing a hyphen are parsed into
 `:'generate-config'` whereas args without them can be directly used as symbols `:generate_config`
 
 It is kind of confusing so please let me know if it does not make sense... 
 
I also have a question about ruby symbols that I thought I should ask.
I've assumed that both kinds of symbols in ruby are similar( `:'abc'` and `:abc` ). But I try to never use the first kind. Do you know of any differences?

Reply:
> hyphen (-) and underscore (_) can be used interchangeably when using them with OptionParser.

ah I didn't know that

> Am I correct in making this assumption? Also, do share if you know of any differences.

I think you're right about the conversion (but maybe it's done in the "compile" time when they are used as literals)